tests/test_dimension.py

```diff
@@ -597,6 +597,16 @@ def test_time_subsampling_fd(self):
                        usave[time_subsampled + 1, x, y],
                        usave[time_subsampled, x, y]]
 
+    def test_issue_1592(self):
```

Comment:
Can we also have the case from the issue.

Reply:
Sure! Just create the operator? Any execution/assertion?

Reply:
somehow assert the generated code is correct?

Reply:
ah, maybe you alreday are! I just see

```
         assert exprs[-1].expr.lhs.indices[0] == IntDiv(time, 2) + 1
```